### PR TITLE
Potential work-around for Rollup cyclic dependency bug

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

There is a bug in Rollup that non-deterministically fails to detect cyclic
dependencies. https://github.com/rollup/plugins/issues/1425
As a result `lru-cache` is sometimes wrapped in a `requireLruCache` function and sometimes it is not.
This makes our build non-deterministic.
`lru-caceh` is not itself part of a circular dependency but `semver/classes/range.js`, which depends on `lru-cache`, is.
By depending on `lru-cache` directly from our app, we change the order in which modules are analyzed by Rollup, hopefully avoiding the buggy behavior.
It probably doesn't matter where we put this import, so I put it as close to the root of the app as I could find.

# Changes

Add `import "lru-cache"` in frontend/src/routes/+layout.ts

# Tests

Inspected generated `vendor` chunk to see that it does not have `hasRequiredLruCache`.
`dfx deploy nns-dapp` locally and did some manual testing.

### CODE REVIEW

**Review by peterpeterparker — APPROVED** [review]
LGTM, well spotted!

File: frontend/src/routes/+layout.ts [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+// dependencies. https://github.com/rollup/plugins/issues/1425
+// As a result 'lru-cache' is sometimes wrapped in a require* function and
+// sometimes it is not. This makes our build non-deterministic.
+// lru-caceh is not itself part of a circular dependency but
+// semver/classes/range.js, which depends on lru-cache, is.
+// By depending on lru-cache directly from our app, we change the order in which
+// modules are analyzed, hopefully avoiding the buggy behavior.
+// It probably doesn't matter where we put this import, so I put it as close to
+// the root of the app as I could find.
+import "lru-cache";
```
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/routes/+layout.ts:11]: Like `// TODO: Remove when the Rollup issue is fixed`?
Or `// TODO: Stop using Rollup and remove this import`?
Or something else?

[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/routes/+layout.ts:11]:
```
// TODO: Remove when the Rollup issue is fixed
```

Sounds good. // TODO are highlighted in my editor, can be handy. Thanks.

**Review by lmuntaner — APPROVED** [review]
Thanks!! 🤞

File: frontend/src/routes/+layout.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/routes/+layout.ts:12]: Done

**Pull request merged**